lib/fastlane/plugin/android_emulator/actions/android_emulator_action.rb

```diff
@@ -36,11 +36,23 @@ def self.run(params)
         end
 
         UI.message("Starting emulator")
-        system("LC_NUMERIC=C; #{sdk_dir}/emulator/emulator @#{params[:name]} -port #{port} > /dev/null 2>&1 &")
-        sh("#{adb} -e wait-for-device")
 
-        until Actions.sh("#{adb} -e shell getprop dev.bootcomplete", log: false).strip == "1" do
-          sleep(5)
+        if params[:locale]          
+          system("LC_NUMERIC=C; #{sdk_dir}/emulator/emulator @#{params[:name]} -no-snapshot -change-locale #{params[:locale]} -port #{port} > /dev/null 2>&1 &")
+          sh("#{adb} -e wait-for-device")
+
+          until Actions.sh("#{adb} -e shell getprop persist.sys.locale", log: false).strip == params[:locale] && 
+            Actions.sh("#{adb} -e shell getprop dev.bootcomplete", log: false).strip == "1" do
+            sleep(5)
+          end
+          sleep(10)
```

Comment:
Why is here an additional 10sec of sleep?

Reply:
The 10 seconds were necessary to change the settings afterwards. Otherwise the emulator was not yet ready. I have now changed the procedure and wait until the sysui_demo_allowed could be completed.